grails-app/views/invoice/_invoiceItems.gsp

```diff
@@ -24,7 +24,12 @@
                         ${invoiceItem?.product?.productCode?:g.message(code:'default.all.label')}
                     </td>
                     <td>
-                        ${invoiceItem?.description}
+                        <g:if test="${invoiceItem?.orderAdjustment}">
+                            ${invoiceItem?.description}
+                        </g:if>
+                       <g:else>
+                           <g:translatedProductName product="${invoiceItem?.product}" />
+                       </g:else>
```

Comment:
Check out the FormatTagLib. I'd prefer to use something like that. We could even add a tag for InvoiceItem. But we at least want to use the following tag 
```
<format:product product="${product}"/>
```
but rewritten to use synonyms instead of the pipe-delimited convention.

Reply:
```
<format:name product="${product}"/>
<format:displayName product="${product}"/>
<format:alternateName product="${product}"/>
...
<format:name object="${invoiceItem.product}"/>
<format:displayName object="${invoiceItem.product}"/>
<format:alternateName product="${product.product}"/>
```